copy/copy.go

```diff
@@ -361,6 +365,44 @@ func supportsMultipleImages(dest types.ImageDestination) bool {
 	return false
 }
 
+// compareImageDestinationManifestEqual compares the `src` and `dest` image's manifests, returning the 'true' if they compare equal.
+// The destination manifest is always fetched.
+func compareImageDestinationManifestEqual(ctx context.Context, options *Options, src types.Image, dest types.ImageDestination) (bool, []byte, string, digest.Digest, error) {
+
+	srcManifest, _, err := src.Manifest(ctx)
+	if err != nil {
+		return false, nil, "", "", errors.Wrapf(err, "Error reading manifest from image")
+	}
+
+	srcManifestDigest, err := manifest.Digest(srcManifest)
+	if err != nil {
+		return false, nil, "", "", errors.Wrapf(err, "Error calculating manifest digest")
+	}
+
+	destImageSource, err := dest.Reference().NewImageSource(ctx, options.DestinationCtx)
+	if err != nil {
+		return false, nil, "", "", nil
+	}
+
+	destManifest, destManifestType, err := destImageSource.GetManifest(ctx, nil)
```

Comment:
Shouldn’t this use `targetInstance`? https://github.com/containers/image/pull/1041#discussion_r499155705

Reply:
oops, thanks! I missed that comment.

I also want to discuss test coverage, but I'll do this in the main PR comment section (so discussion doesn't get lost when this comment becomes outdated )